tests/time_tests/test_runner/conftest.py

```diff
@@ -0,0 +1,81 @@
+"""
+Basic high-level plugin file for pytest.
+
+See [Writing plugins](https://docs.pytest.org/en/latest/writing_plugins.html)
+for more information.
+
+This plugin adds the following command-line options:
+
+* `--test_conf` - Path to test configuration file. Used to parametrize tests.
+  Format: YAML file.
+* `--exe` - Path to a timetest binary to execute.
+* `--niter` - Number of times to run executable.
+"""
+
+# pylint:disable=import-error
+import pytest
+from pathlib import Path
+import yaml
+
+
+# -------------------- CLI options --------------------
+
+def pytest_addoption(parser):
+    """Specify command-line options for all plugins"""
+    parser.addoption(
+        "--test_conf",
+        type=Path,
+        help="Path to test config",
+        default=Path(__file__).parent / "test_config.yml"
+    )
+    parser.addoption(
+        "--exe",
+        required=True,
+        dest="executable",
+        type=Path,
+        help="Path to a timetest binary to execute",
+    )
+    parser.addoption(
+        "--niter",
+        type=int,
+        help="Number of iterations to run executable and aggregate results",
+        default=3
+    )
+    # TODO: add support of --mo, --omz etc. required for OMZ support
+
+
+@pytest.fixture(scope="session")
+def test_conf(request):
+    """Fixture function for command-line option."""
+    return request.config.getoption('test_conf')
+
+
+@pytest.fixture(scope="session")
+def executable(request):
+    """Fixture function for command-line option."""
+    return request.config.getoption('executable')
+
+
+@pytest.fixture(scope="session")
+def niter(request):
+    """Fixture function for command-line option."""
+    return request.config.getoption('niter')
+
+# -------------------- CLI options --------------------
+
+
+def pytest_generate_tests(metafunc):
+    """Pytest hook for test generation.
+
+    Generate parameterized tests from discovered modules and test config
+    parameters.
+    """
+    with open(metafunc.config.getoption('test_conf'), "r") as file:
+        test_cases = yaml.load(file, Loader=yaml.FullLoader)
+    if test_cases:
+        metafunc.parametrize("instance", test_cases)
+
+
+def pytest_make_parametrize_id(config, val, argname):
+    """Pytest hook for user-friendly test name representation"""
+    return " {0}:{1} ".format(argname, val)
```

Comment:
I'm thinking that argname can be eliminated for shorter test name. What do you think?

Reply:
Done. Now keys of all internal dicts will be eliminated. Example: device: {name: CPU} -> device_CPU